openedx/adg/admin_dashboard/admin_settings/tests/test_views.py

```diff
@@ -13,75 +13,72 @@
-    create_admin_accounts(total_admins, BU_ADMIN, False)
-    assert AdminAccountListView().get_queryset().count() == expected_count
-
-
-@pytest.mark.parametrize('total_admins, expected_count', [
-    (5, 5),
-    (10, 10),
-])
-@pytest.mark.django_db
-@skip_unless_lms
-def test_admin_accounts_context_data(total_admins, expected_count, super_user):
-    """
-    Test context data for `AdminAccountListView`.
-    """
-    create_admin_accounts(total_admins, ADG_ADMIN, True)
+    _, _, adg_admins, _ = create_admin_accounts(total_admins, ADG_ADMIN, True)
+    _, _, bu_admins, _ = create_admin_accounts(total_admins, BU_ADMIN, True)
 
     url = reverse('admin_accounts')
-    admin_accounts_get_request = RequestFactory().get(url)
-    admin_accounts_get_request.user = super_user
+    admin_accounts_get_request = request_factory.get(url)
+
+    admin = adg_admins[0] if user_type == ADG_ADMIN else bu_admins[0]
+    admin_accounts_get_request.user = admin
 
     response = AdminAccountListView.as_view()(admin_accounts_get_request)
     actual_admin_list = response.context_data['admins']
 
     assert len(actual_admin_list) == expected_count
 
 
-@pytest.mark.parametrize('total_invitations, expected_count', [
-    (5, 5),
-    (10, 10),
+@pytest.mark.parametrize('total_invitations, user_type, expected_count', [
+    (5, ADG_ADMIN, 10),
+    (10, ADG_ADMIN, 10),
+    (5, BU_ADMIN, 5),
+    (10, BU_ADMIN, 10),
 ])
 @pytest.mark.django_db
 @skip_unless_lms
-def test_invitations_list_queryset(total_invitations, expected_count):
-    """
-    Test queryset for `PendingInvitationListView` returns expected number of pending invitations.
-    """
-    InvitationFactory.create_batch(total_invitations, status=Invitation.PENDING)
-    InvitationFactory.create_batch(total_invitations, status=Invitation.ACCEPTED)
-    assert PendingInvitationListView().get_queryset().count() == expected_count
-
-
-@pytest.mark.parametrize('total_invitations, expected_count', [
-    (5, 5),
-    (10, 10),
-])
-@pytest.mark.django_db
-@skip_unless_lms
-def test_invitations_list_context_data(total_invitations, expected_count, super_user):
+def test_invitations_list_context_data(
+    total_invitations, user_type, expected_count, adg_admin, business_line, request_factory
+):
     """
     Test context_data for `PendingInvitationListView`.
     """
-    InvitationFactory.create_batch(total_invitations, status=Invitation.PENDING)
+    InvitationFactory.create_batch(
+        total_invitations,
+        status=Invitation.PENDING,
+        permission=Invitation.BU_ADMIN,
+        business_unit=business_line,
+    )
+    InvitationFactory.create_batch(
+        total_invitations,
+        status=Invitation.PENDING,
+        permission=Invitation.ADG_ADMIN,
+    )
+
+    if user_type == ADG_ADMIN:
+        admin = adg_admin
+    else:
+        admin = UserFactory(is_active=True, is_staff=True)
+        admin.groups.add(business_line.group)  # pylint: disable=no-member
```

Comment:
Can we make a fixture for this?

Reply:
Actually I thought of this but we need business line with in that fixture so I am short of ideas like how to manage that thing because its simple for ADG Admins but not in case of BU admins.

Reply:
How about creating a business line in the fixture and adding it to the user? That way we won't need to add the `business_line` fixture. We can instead use `bu_admin.business_line` wherever we need it